### PR TITLE
docs(jsdoc): add and polish

### PR DESCRIPTION
**Pull request by JacopoDaeli**

### CODE REVIEW

File: lib/plugins/cdn.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ const CDN_BUCKET = 'warehouse-cdn';
  */
 
 module.exports = fp(
+  /**
+   * Initialize CDN plugin
+   * @param {import('../warehouse').WarehouseApp} fastify
```
Comment (decompil3d) [inline lib/plugins/cdn.js:28]: Why is this type different than the `fastify` param in other plugins?

Comment (JacopoDaeli) [reply lib/plugins/cdn.js:28]: https://github.com/godaddy/warehouse.ai/pull/112/files#diff-642dfcaadcf076d1d0bff936dcb0ac62ab947e6afa45e2356a1d31d2bb67cd8dR14

Comment (JacopoDaeli) [reply lib/plugins/cdn.js:28]: It's an extension where I declare that there are also 2 properties call `dynamo` and `s3`.

**Comment by JacopoDaeli** [comment]
@decompil3d I have eslint saying that there are syntax error in my JSDoc. I cannot see them. Do you see any?

File: lib/config.js [hunk 2]
```diff
@@ -1,6 +1,12 @@
 const fp = require('fastify-plugin');
 
 module.exports = fp(
+  /**
+   * Initialize config
+   * @param {import('fastify').FastifyInstance} fastify
```
Comment (decompil3d) [inline lib/config.js:6]: eslint does not understand `import` calls in JSDoc. You can work around by using a `typedef`:

```js
/**
 * @typedef {import('fastify').FastifyPluginOptions} FastifyPluginOptions
 */
/**
 * @param {FastifyPluginOptions} opts
 */
```

**Pull request merged**